src/Symfony/Bridge/Doctrine/PropertyInfo/DoctrineExtractor.php

```diff
@@ -169,11 +171,12 @@ public function getTypes($class, $property, array $context = [])
                     break;
                 case Type::BUILTIN_TYPE_ARRAY:
                     switch ($typeOfField) {
-                        case DBALType::TARRAY:
-                        case DBALType::JSON_ARRAY:
+                        case $useDeprecatedConstants ? DBALType::TARRAY : Types::ARRAY:
+                        case 'json_array':
```

Comment:
`Type::JSON_ARRAY` and `Types::JSON_ARRAY` are both deprecated, this is why we need to hardcode it now.